### PR TITLE
[Event Hubs] Update system property types

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

 The focus of these changes is to update the type of several existing values in the `EventData.SystemProperties` collection so that they are properly represented as .NET string types.  Previously,
 the underlying AMQP types were unintentionally returned, forcing callers to call `ToString()` to read the value.  This is a behavioral breaking change that will impacts only those callers who were explicitly casting system property values to `AmqpAddress` or  `AmqpMessageId` before calling `ToString`.

  The affected system properties are:
  - MessageId
  - CorelationId
  - To
  - ReplyTo

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpSystemProperties.cs [hunk 1]
```diff
@@ -130,7 +130,12 @@ internal class AmqpSystemProperties : IReadOnlyDictionary<string, object>
 
                 if (_amqpMessage.HasSection(AmqpMessageSection.MessageAnnotations))
                 {
-                   return _amqpMessage.MessageAnnotations[key];
+                    return _amqpMessage.MessageAnnotations[key] switch
+                    {
+                        AmqpMessageId id => id.ToString(),
```
Comment (JoshLove-msft) [inline sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpSystemProperties.cs:135]: Do we need to guard against null values here?

Comment (JoshLove-msft) [reply sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpSystemProperties.cs:135]: Nevermind..

**Review by JoshLove-msft — APPROVED** [review]

**Review by christothes — APPROVED** [review]

**Review by m-redding — APPROVED** [review]

File: sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,14 @@
 
 ### Breaking Changes
 
+- The type of several existing values in the `EventData.SystemProperties` collection have been changed so that they are properly represented as .NET string types.  Previously, the underlying AMQP types were unintentionally returned, forcing callers to call `ToString()` to read the value.  
+
+  This is a behavioral breaking change that will impacts only those callers who were explicitly casting system property values to `AmqpAddress` or `AmqpMessageId` before calling `ToString`.   The affected system properties are:
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs/CHANGELOG.md:11]:
```suggestion
  This is a behavioral breaking change that will impacts only those callers who were explicitly casting system property values to `AmqpAddress` or `AmqpMessageId` before calling `ToString()`.   The affected system properties are:
```

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,14 @@
 
 ### Breaking Changes
 
+- The type of several existing values in the `EventData.SystemProperties` collection have been changed so that they are properly represented as .NET string types.  Previously, the underlying AMQP types were unintentionally returned, forcing callers to call `ToString()` to read the value.  
+
+  This is a behavioral breaking change that will impacts only those callers who were explicitly casting system property values to `AmqpAddress` or `AmqpMessageId` before calling `ToString`.   The affected system properties are:
```
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/CHANGELOG.md:11]:
```suggestion
  This is a behavioral breaking change that will impacts only those callers who were explicitly casting system property values to `AmqpAddress` or `AmqpMessageId` before calling `ToString()`.   The affected system properties are:
```

**Pull request merged**